File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.py
def solution(number, limit, power):
    answer = 0
    
    for i in range(1, number + 1):
        count = 0
        for j in range(1, int(i ** 0.5) + 1):
            if j**2 == i:
                count += 1
            elif i % j == 0:
                count += 2
        if count <= limit:
            answer += count
        else:
            answer += power   
    
    return answer

File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.java
class Solution {
    public int countDivisors(int num) {
        int count = 0;
        
        for (int i = 1; i*i <= num; i++) {
            if (num % i == 0) {
                count++;
                if (i != num/i) {
                    count++;
                }
            }
        }
        return count;
    }
    
    
    public int solution(int number, int limit, int power) {
        int answer = 0;
        
        for (int j = 1; j <= number; j++) {
            int divCount = countDivisors(j);
            if (divCount <= limit) {
                answer += divCount;
            }
            else {
                answer += power;
            }
        }
        
        return answer;
    }
}